hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/PmemVolumeManager.java

```diff
@@ -68,15 +71,19 @@
      *                      -1 if we failed.
      */
     long reserve(long bytesCount) {
-      while (true) {
-        long cur = usedBytes.get();
-        long next = cur + bytesCount;
-        if (next > maxBytes) {
-          return -1;
-        }
-        if (usedBytes.compareAndSet(cur, next)) {
-          return next;
+      if(bytesCount > 0) {
+        while (true) {
+          long cur = usedBytes.get();
+          long next = cur + bytesCount;
+          if (next > maxBytes) {
+            return -1;
+          }
+          if (usedBytes.compareAndSet(cur, next)) {
+            return next;
+          }
         }
+      } else {
```

Comment:
the else logic can change this method's original behavior. This method should return a positive value or -1.